code-challenges.test.js

```diff
@@ -15,42 +16,132 @@
+        const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+        const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+      expect(newArray(fullArr1)).toEqual([-9, 7, 9, 199])
+      expect(newArray(fullArr2)).toEqual([-823, 7, 23])
+    })
+  })
 
-// b) Create the function that makes the test pass.
+// Succesful failure 
+// ● newArray › takes in an array and returns a new array of only odd numbers sorted from least to greatest.
+// ReferenceError: newArray is not defined
 
 
-// --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+// b) Create the function that makes the test pass.
+// Pesudo Code 
+// create a function: oddNumbers 
+// input: array given 
+// returns a new array with only odd numbers sorted from least to greatest 
+// will need to use .filter() to return a new array with only odd numbers 
+// our returned odd numbers need to be ordered from least to greatest -->use .sort()
+
+
+const oddNumbers = (array) => {
+    let odds = []
+    array.filter(value => value % 2 !== 0)
+    return odds.filter(value  => typeof value === 'numbers')
+}
+// console.log(oddNumbers(fullArr1));
+
+// I got stuck on the code above and could not figure out how to create the function that would return an array of only odd numbers sorted from least to greatest. I got my code to return all the odd values initially but the output was also including strings and numbers that it recognized as odd values. I then attempted to add the typeof method to return only the odd numbers but for some reason my code is not running, and I want to say it is due to the syntax and structure of my function? I also created the code below as a reference to how to obtain only numbers in our array but could not figure out where else to go with my code in order to get the result the question is asking. Would a for loop be a better approach here? 
+
+const oddNumbers2 = (array) => {
+    return array.filter(value => typeof value === 'number')
+}
+// console.log(oddNumbers2(fullArr1));
+        
+        
+        
+        // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
 
-const numbersToAdd2 = [0, 7, -8, 12]
-// Expected output: [0, 7, -1, 11]
 
-const numbersToAdd3 = []
-// Expected output: []
+describe ("cumulativeSum", () => {
+    it ("takes in an array and returns an array of the accumulating sum", () => {
+        const numbersToAdd1 = [2, 4, 45, 9]
+        const numbersToAdd2 = [0, 7, -8, 12]
+        const numbersToAdd3 = []
+      expect(anotherNewArray(numbersToAdd1)).toEqual([2, 6, 51, 60])
```

Comment:
Your expect statements are doing function calls; therefore, you should be calling the same function you declared on Line 110 `cumulativeSum`